components/geo-map.client.vue

```diff
@@ -37,7 +61,62 @@ const context: GeoMapContext = {
 	},
 };
 
-function updateMarkers() {
+const ptDistanceSq = function (pt1: LeafletPoint, pt2: LeafletPoint): number {
+	let dx = pt1.x - pt2.x;
+	let dy = pt1.y - pt2.y;
+	return dx * dx + dy * dy;
+};
+
+// Bind a popup listing nearby data to markers close to each other.
+// Remove existing if there are no nearby markers on the map any more.
+const addNearbyDataPopup = function (marker) {
+	const featureGroup = context.featureGroups.markers;
+	const map = context.map;
+	if (featureGroup === null || map === null) return;
+
+	let distance = Math.floor(2 * map.getZoom());
+	let nearbyMarkerData = [];
+	const pxSq = distance * distance;
+	const markerPt = map.latLngToLayerPoint(marker.getLatLng());
+
+	const id = featureGroup.getLayerId(marker);
+	Object.values(featureGroup.getLayers()).forEach((m: LeafletMarker) => {
+		if (map.hasLayer(m)) {
+			const mPt = map.latLngToLayerPoint(m.getLatLng());
+			if (ptDistanceSq(mPt, markerPt) < pxSq) {
+				nearbyMarkerData.push(m.feature);
+			}
+		}
+	});
+
+	if (nearbyMarkerData.length > 1) {
+		const markers = nearbyMarkerData.sort((a, b) => {
+			return a.properties.label.localeCompare(b.properties.label);
+		});
+
+		// @todo determine whether grouping is needed based on the number of
+		// feature groups once separate feature groups for queries are supported.
+		let contentTypes = [];
+		markers.forEach((marker) => {
+			if (!contentTypes.includes(marker.properties.targetType)) {
+				contentTypes.push(marker.properties.targetType);
+			}
+		});
+
+		componentPopups.value.push({
+			id: id as string,
+			props: {
+				markers: markers,
+				groupMarkers: contentTypes.length > 1,
+			},
+		});
+	} else if (marker.getPopup()) {
+		console.log(marker);
+		marker.removePopup();
```

Comment:
I couldn't find the `removePopup()` method in the leaflet documentation, but also wasn't able to trigger this condition

Reply:
right, that seems unnecessary as markers are redrawn. Good catch!

Reply:
so, will you keep working on this branch or should I merge and delete it?

Reply:
Yes, I should just remove that.